toml.md

```diff
@@ -293,9 +298,14 @@ For convenience, some popular characters have a compact escape sequence.
 ```
 
 Any Unicode character may be escaped with the `\xHH`, `\uHHHH`, or `\UHHHHHHHH`
-forms. The escape codes must be valid Unicode
+forms. The escape codes must be Unicode
 [scalar values](https://unicode.org/glossary/#unicode_scalar_value).
 
+Note that there is no low-level byte sequence syntax in TOML, and that the
+character escape codes must not be made to work in such a way. So for binary
```

Comment:
I'm not quite happy with the sentence: "Note that there is no low-level byte sequence syntax in TOML, and that the character escape codes must not be made to work in such a way."

I get what you're trying to say, but what exactly is the meaning of that "must"? Would be TOML police come if you try to do so anyway? Hardly! If one really wants to do so, one could treat Unicode strings that contain just code points between U+0000 and U+00FF as byte sequences. It wouldn't make much sense to use `\xHH` to represent otherwise unwriteable bytes in such sequences, since it would entail an overhead of up to 300%, while Base64 has an overhead of only about 33%. Still, there is no law against it.

Therefore I would propose to replace these two lines with something like: "Keep in mind that all TOML strings are sequences of Unicode characters, *not* of bytes. For binary" – and then the next lines, recommending "Base64 or another binary-to-text encoding", could neatly follow.

Reply:
> If one really wants to do so, one could treat Unicode strings that contain just code points between U+0000 and U+00FF as byte sequences.

Heh, actually that doesn't work for the byte range 80 to FF, because in UTF-8, `"\x80"` through `"\xFF"` are each two-byte code point.

But @ChristianSi, I like the approach that you are taking. I will keep your first sentence about the same, but it will lead into a discussion on byte sequences instead. This allows the text to flow a little more naturally. The rest is reformulated to emphasize that the methods used to get bytes from strings and vice versa are not a part of TOML, but rather external methods. I also mention hexadecimal strings as such a method, since those show up a lot as hashes nowadays. I linked to a [Binary64](https://www.base64decode.org/) site describing that method.

Here's the most recent version of the paragraph. Would this work?

```md
Keep in mind that all TOML strings are sequences of Unicode characters, _not_
byte sequences. For binary data, avoid using these escape codes. Instead,
external binary-to-text encoding strategies, like hexadecimal sequences or
[Base64](https://www.base64decode.org/), are recommended for converting between
bytes and strings.
```

Reply:
That’s a big improvement, I’d go for it 👍